src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,171 @@
+import java.util.Scanner;
+
+import duke.dukeexception.DateTimeFormatException;
+import duke.dukeexception.DukeException;
+import duke.dukeexception.IncomplateCommandException;
+import duke.dukeexception.NoSuchCommandException;
+import duke.dukeexception.TaskCompletionException;
+import duke.dukeexception.TaskOutOfBoundException;
+import duke.parser.CommandType;
+import duke.parser.Parser;
+import duke.storage.Cache;
+import duke.storage.TaskList;
+import duke.tasks.Task;
+
+/**
+ * The main class for personal assistant bot Duke.
+ */
+public class Duke {
+    private static TaskList taskList;
+
+    public Duke() {
+        taskList = new TaskList();
+    }
+
+    /**
+     * Launches the bot and says hello. Reads in the previous task list in cache and take user inputs to.
+     * @param args User inputs.
+     * @throws IOException
+     * @throws DukeException
+     */
+    public static void main(String[] args) throws IOException, DukeException {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+        Cache cache = new Cache(System.getProperty("user.dir") + "\\data\\duke.txt");
+        taskList = cache.printPath();
+        run();
+        cache.update(taskList);
+    }
+
+    /**
+     * Executes commands:
+     * 1. Greetings on opening.
+     * 2. todo [task name]: to add a ToDo Task.
+     * 3. deadline [task name] /by yyyy-MM-dd: to add a Deadline task.
+     * 4. event [task name] /at yyyy-MM-dd hh:mm: to add a Event task.
+     * 5. bye: to exit.
+     * 6. mark [index]: to mark done for the task with given index.
+     * 7. unmark [index]: to unmark done for the task with given index.
+     * 8. delete [index]: to delete the task with given index.
+     * 9. find [String]: to find the corresponding task.
+     * 10. list: to list all ongoing tasks.
+     */
+    public static void run() throws IncomplateCommandException, TaskOutOfBoundException,
+            NoSuchCommandException, TaskCompletionException, DateTimeFormatException {
+        Scanner in = new Scanner(System.in);
+        boolean lastCommandOrNot = false;
+        Task task;
+        String command;
+        String suggestion;
+        String[] commandList;
+        Integer taskIndex = null;
+        Integer i;
+        CommandType type;
+
+        Ui.greet();
+        while (!lastCommandOrNot) {
+            command = in.nextLine().trim();
+            commandList = command.split(" ", 2);
+            command = (Parser.multipleVariable(commandList[0])) ? commandList[0] : command;
+            type = CommandType.map(command);
+
+            try {
+                switch (type) {
+                case BYE: {
+                    Ui.bye();
+                    lastCommandOrNot = true;
+                    break;
+                }
```

Comment:
Statement within case blocks do not need curly braces according to the coding standards.
Is it possible to remove them?

Reply:
Oh thanks for reminder, I have made the changes!